src/sincere_singularities/utils.py

```diff
@@ -51,3 +52,17 @@ def load_json(filename: str, json_type: type[T]) -> T:
         typed_json: T = dacite.from_dict(json_type, loaded_json)
 
         return typed_json
+
+
+def check_similarity(first: str, second: str) -> float:
+    """
+    Measure of the strings' similarity as a float.
+
+    Args:
+        first (str): The first string.
+        second (str): The second string.
+
+    Returns:
+        float: The similarity of the two strings [0, 1]
+    """
+    return difflib.SequenceMatcher(None, first, second).ratio()
```

Comment:
Sequence Matcher wont work.
It uses the [Gestalt Pattern Matching Algorithm](https://en.wikipedia.org/wiki/Gestalt_pattern_matching).
So it doesnt actually use any Sementic Analysis/Sentence-Matching AI, but just math.

For example: with these two sentences:
```py
>>> str0 = "The cat is on the mat."
>>> str1 = "A feline rests atop a rug."
```
(These sentences are very unrelated to the project, but show the point. Sentences generated with ChatGPT.)

I get a SequenceMatcher result of:
```py
>>> difflib.SequenceMatcher(None, str1, str0).ratio()
0.2127659574468085
```

But for example [Huggingface Sentence-Similarity Demo](https://huggingface.co/tasks/sentence-similarity) gives me 0.57, which is much closer.
So we have to use a (sentence-transformer AI for this...)

Reply:
I'll leave that up to you.